### PR TITLE
chore(BaseInput): reusable, internal Input component

### PR DESCRIPTION
**Pull request by YossiSaadi**
https://monday.monday.com/boards/3532714909/pulses/6437205266

### CODE REVIEW

File: packages/core/src/components/BaseInput/BaseInput.tsx
Comment (talkor) [inline packages/core/src/components/BaseInput/BaseInput.tsx:35]: What's this?

Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/BaseInput.tsx:35]: oops :)

File: packages/core/src/components/BaseInput/BaseInput.tsx
Comment (talkor) [inline packages/core/src/components/BaseInput/BaseInput.tsx:1]: I'd call it just Input

File: packages/core/src/components/BaseInput/BaseInput.types.ts
Comment (talkor) [inline packages/core/src/components/BaseInput/BaseInput.types.ts:1]: Yes yes yes!!

File: packages/core/src/components/BaseInput/BaseInput.module.scss
Comment (talkor) [inline packages/core/src/components/BaseInput/BaseInput.module.scss:10]: Let's make it a standard?

File: packages/core/src/components/BaseInput/BaseInput.module.scss
Comment (talkor) [inline packages/core/src/components/BaseInput/BaseInput.module.scss:41]: Necesarry?

Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/BaseInput.module.scss]: The PR was still in draft state, copied some of the logic from `TextField`, removed unnecessary stuff now

File: packages/core/src/components/BaseInput/BaseInput.module.scss
Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/BaseInput.module.scss:10]: wdym?

File: packages/core/src/components/BaseInput/BaseInput.tsx
Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/BaseInput.tsx:1]: Hmm let's discuss it
I'm leaving it open for having an Input component which would be exported from Vibe

**Comment by YossiSaadi** [comment]
added ability to hide stories here #2062 
make sure to merge this only after #2062 is merged

File: packages/core/src/components/BaseInput/BaseInput.module.scss
Comment (talkor) [reply packages/core/src/components/BaseInput/BaseInput.module.scss:10]: Logical properties

Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/BaseInput.module.scss:10]: hmm I agree
Chrome 87+ support
we don't have users with less than 90, but anyway when we upgrade browsers it should be supported

File: packages/core/src/components/BaseInput/BaseInput.module.scss
Comment (talkor) [inline packages/core/src/components/BaseInput/BaseInput.module.scss:79]: Didn't we say we can't use `has` for now?

File: packages/core/src/components/BaseInput/BaseInput.tsx
Comment (talkor) [inline packages/core/src/components/BaseInput/BaseInput.tsx:17]: Can they get two different roles?

File: packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx
Comment (talkor) [inline packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx:21]: I think you can use `container` or something similar to get the outer element

File: packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx
Comment (talkor) [inline packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx:82]: Extract to a variable?

File: packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx
Comment (talkor) [inline packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx:85]: Nice!

**Review by talkor — APPROVED** [review]
Very cool!

File: packages/core/src/components/BaseInput/BaseInput.module.scss
Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/BaseInput.module.scss]: added support for both browsers that do and do not support has

File: packages/core/src/components/BaseInput/BaseInput.tsx
Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/BaseInput.tsx:16]: yep, for example in combobox, the wrapper should be "search" and the input "combobox" (it sounds weird, I know)

File: packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx
Comment (YossiSaadi) [reply packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx:21]: container gives where the BaseInput is rendered, so I need to query the there the `firstChild` so it is pretty much the same..
I'll leave it the way it is I think

**Pull request merged**
